homeassistant/components/plugwise/config_flow.py

```diff
@@ -20,14 +31,77 @@
 from homeassistant.helpers.typing import DiscoveryInfoType
 
 from .const import (  # pylint:disable=unused-import
+    CONF_USB_PATH,
     DEFAULT_PORT,
     DEFAULT_SCAN_INTERVAL,
     DOMAIN,
+    FLOW_NET,
+    FLOW_TYPE,
+    FLOW_USB,
+    PW_TYPE,
+    SMILE,
+    STICK,
+    STRETCH,
     ZEROCONF_MAP,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
+CONF_MANUAL_PATH = "Enter Manually"
+
+CONNECTION_SCHEMA = vol.Schema(
+    {
+        vol.Required(FLOW_TYPE, default=FLOW_NET): vol.In(
+            {FLOW_NET: f"Network: {SMILE} / {STRETCH}", FLOW_USB: "USB: Stick"}
+        ),
+    },
+)
+
+
+@callback
+def plugwise_stick_entries(hass):
+    """Return existing connections for Plugwise USB-stick domain."""
+    sticks = []
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data.get(PW_TYPE) == STICK:
+            sticks.append(entry.data.get(CONF_USB_PATH))
+    return sticks
+
+
+async def validate_usb_connection(self, device_path=None) -> Dict[str, str]:
+    """Test if device_path is a real Plugwise USB-Stick."""
+    errors = {}
+    # Avoid creating a 2nd connection to an already configured stick
+    if device_path in plugwise_stick_entries(self):
+        errors[CONF_BASE] = "already_configured"
+        return errors, None
+
+    stick = await self.async_add_executor_job(plugwise.stick, device_path)
```

Comment:
Is it really necessary to call a class using `async_add_executor_job()`?

Maybe better to use `from plugwise import stick`
and then use something like `api_stick = stick(device_path)`?